docs/provider-setup.md

```diff
@@ -85,9 +91,34 @@ domain = "http://192.168.56.102"
 #no_passphrase = true
 ```
 
-with suitable replacements.
+with suitable [replacements](#provider-options).
 
 Create the folders:
 ```(shell)
 curl http://192.168.56.102/cgi-bin/csaf_provider.go/create
 ```
+Or using the uploader:
+```(shell)
+./csaf_uploader -a create -u http://192.168.56.102/cgi-bin/csaf_provider.go
+```
+
+## Provider options
+Provider has many config options described as following:
+
+ - password: Authentication password for accessing the CSAF provider.
+ - key: The private OpenPGP key.
+ - folder: Specify the root folder. Default: `/var/www/`.
+ - web: Specify the web folder. Default: `/var/www/html`.
+ - tlps: Set the allowed TLP comming with the upload request (one or more of "csaf", "white", "amber", "green", "red").             
+   The "csaf" selection lets the provider takes the value from the CSAF document.         
+   These affects the list items in the web interface.      
+   Default: `["csaf", "white", "amber", "green", "red"]`.
+ - upload_signature: Send signature with the request, an additional input-field in the web interface will be shown to let user enter an ascii armored signature. Default: `false`.
+ - openpgp_url: URL to OpenPGP key-server. Default: `https://openpgp.circl.lu`.
+ - domain: The domain name. Default: `http://$SERVER_NAME`.
```

Comment:
@s-l-teichmann the `domain` config options is a bit unbalanced. It is called domain, but in the code, the value will be taken as base_url. And then there is `http:` which is okay for development, but not as default I guess. What about renaming it to serve_url and then default it to `https://$SERVER_NAME´ or so. Where does the name `SERVER_NAME` come from? My search engine does not tell me it is a standard environment variable. For development either `localhost` or `$hostname` would be alternatives. What do you think?

Reply:
@s-l-teichmann ping

Reply:
@bernhardreiter  We could use `https://` as a default, sure.
Concerning `domain` ... my bike shed opinion would more tend to something like `canonical_url_prefix`.

Reply:
Okay, changed to `canonical_url_prefix`.